pkg/agent/monitoraddnodes.go

```diff
@@ -180,36 +197,47 @@ func (mon *addNodeMonitor) nodeHasJoinedClusterAndIsReady() (bool, bool, error)
 }
 
 func (mon *addNodeMonitor) logCSRsPendingApproval(signerName string) {
-	// TODO? The CSRs have no IP address to identify for which
-	// node it is for, so it is possible to log CSRs pending for
-	// other nodes.
+	csrsPendingApproval := mon.getCSRsPendingApproval(signerName)
+
+	for _, csr := range csrsPendingApproval {
+		mon.logStatus(fmt.Sprintf("CSR %s with signerName %s and username %s is Pending and awaiting approval",
+			csr.Name, csr.Spec.SignerName, csr.Spec.Username))
+	}
+}
+
+func (mon *addNodeMonitor) clusterHasFirstCSRPending() bool {
+	return len(mon.getCSRsPendingApproval(firstCSRSignerName)) > 0
+}
+
+func (mon *addNodeMonitor) clusterHasSecondCSRPending() bool {
+	return len(mon.getCSRsPendingApproval(secondCSRSignerName)) > 0
+}
+
+func (mon *addNodeMonitor) getCSRsPendingApproval(signerName string) []certificatesv1.CertificateSigningRequest {
 	csrs, err := mon.cluster.API.Kube.ListCSRs()
 	if err != nil {
 		logrus.Debugf("error calling listCSRs(): %v ", err)
 	}
 
+	var matchedCSRs []certificatesv1.CertificateSigningRequest
 	for _, csr := range csrs.Items {
 		if len(csr.Status.Conditions) == 0 {
 			// CSR is Pending and awaiting approval
-			if signerName != "" && signerName != csr.Spec.SignerName {
-				continue
+			if csr.Spec.SignerName == signerName {
```

Comment:
nit: to reduce the nesting depth level, the previous happy path conditions was better for me